File: COUNT-HILLS-VALLEYS/chv.java
class Solution {
    public int countHillValley(int[] nums) {
        List<Integer> arr = new ArrayList<>();
        arr.add(nums[0]);

        for (int i : nums) {
            if (arr.get(arr.size() - 1) != i) {
                arr.add(i);
            }
        }

        int count = 0;
        for (int j = 1; j < arr.size() - 1; j++) {
            if (arr.get(j) > arr.get(j - 1) && arr.get(j) > arr.get(j + 1)) {
                count++;
            }
            if (arr.get(j) < arr.get(j - 1) && arr.get(j) < arr.get(j + 1)) {
                count++;
            }
        }

        return count;
    }
}

File: COUNT-HILLS-VALLEYS/chv.cpp
class Solution {
public:
    int countHillValley(vector<int>& nums) {
        vector <int> arr;
        arr.push_back(nums[0]);
        for (int i : nums){
            if (arr[arr.size() - 1] != i){
                arr.push_back(i);
            }
        }
        int count = 0;
        for (int j = 1; j < arr.size() - 1; j++){
            if(arr[j] > arr[j + 1] && arr[j] > arr[j - 1]){
                count ++;
            }
            if(arr[j] < arr[j + 1] && arr[j] < arr[j - 1]){
                count ++;
            }
        }
        return count;
    }
};